code/modules/psionics/faculties/coercion.dm

```diff
@@ -248,21 +264,18 @@
 			to_chat(user, SPAN_CULT("Not even a psion of your level can speak to the dead."))
 			return
 
-		var/obj/item/organ/internal/augment/psi/psiaug
-		for (psiaug in target)
-			if(psiaug && !psiaug.is_broken())
-				break
-		if (!psiaug)
+		var/mob/living/carbon/C = target
```

Comment:
You haven't checked if this is a carbon

Reply:
Also, I still don't like how you're now copypasting this psiaug stuff all over the place. Please just make it into its own proc that returns TRUE if aug is present or if the creature can commune and FALSE if it can't.